test/distributed/test_c10d_common.py

```diff
@@ -1378,11 +1385,20 @@ class CommResult:
     _work: torch.classes.c10d.Work
 
 
-def wrap_comm_result(result):
+def wrap_comm_result(result: Tuple[Any]) -> Tuple[Any]:
+    def wrap(work, e):
+        assert isinstance(e, torch.Tensor), (
+            "Excepting collection of tensors as the first element in the "
+            "return value of communication operations."
+        )
+
+        return CommResult(e, work)
+
+    # E.g.,
     # allreduce_ returns ([tensor], work)
-    tensor = result[0][0]
+    # allgather_ returns ([[tensor1, tensor2]], work)
```

Comment:
any reason we return an additional [] wrapper?

Reply:
it's the existing signature of allgather. When allgather was developed back in 2018, it supports single-process multi-device use cases. The outer list is for the multi-device scenarios, where which inner-list represent one device. Same thing applies to allreduce as well.